lib/db/schema/patch-086-087.sql

```diff
@@ -0,0 +1,62 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('86');
+
+CREATE PROCEDURE `verifyTokensWithMethod_3` (
+  IN `tokenIdArg` BINARY(32),
+  IN `verificationMethodArg` INT,
+  IN `verifiedAtArg` BIGINT(1)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+    -- Update session verification methods
+    UPDATE `sessionTokens` SET verificationMethod = verificationMethodArg, verifiedAt = verifiedAtArg
+    WHERE tokenId = tokenIdArg;
+
+END;
+
+CREATE PROCEDURE `verifyTokenCode_2` (
+  IN `tokenVerificationCodeHashArg` BINARY(32),
+  IN `uidArg` BINARY(16)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  SET @tokenVerificationId = NULL;
+  SELECT tokenVerificationId INTO @tokenVerificationId FROM unverifiedTokens
+    WHERE uid = uidArg
+    AND tokenVerificationCodeHash = tokenVerificationCodeHashArg
+    AND tokenVerificationCodeExpiresAt >= (UNIX_TIMESTAMP(NOW(3)) * 1000);
+
+  IF @tokenVerificationId IS NULL THEN
+    SET @expiredCount = 0;
+    SELECT COUNT(*) INTO @expiredCount FROM unverifiedTokens
+      WHERE uid = uidArg
+      AND tokenVerificationCodeHash = tokenVerificationCodeHashArg
+      AND tokenVerificationCodeExpiresAt < (UNIX_TIMESTAMP(NOW(3)) * 1000);
+
+    IF @expiredCount > 0 THEN
+      SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 2101, MESSAGE_TEXT = 'Expired token verification code.';
+    END IF;
+  END IF;
+
+  UPDATE securityEvents
+  SET verified = true
+  WHERE tokenVerificationId = @tokenVerificationId
+  AND uid = uidArg;
+
+  DELETE FROM unverifiedTokens
```

Comment:
I simplified this by removing the transaction portion, which will return the `affectedRows` from the `DELETE`. I don't remember the original logic why I put this in a transaction but think it is ok to remove.

Reply:
What a pain. It does kinda wanna be a transaction (I logged it, I did it), but...